src/de.rs

```diff
@@ -348,7 +373,18 @@ impl<'de> de::MapAccess<'de> for PyMappingAccess<'de> {
         K: de::DeserializeSeed<'de>,
     {
         if self.key_idx < self.len {
-            let mut item_de = Depythonizer::from_object(self.keys.get_item(self.key_idx)?);
+            let key = self.keys.get_item(self.key_idx)?;
+            let py = key.py();
+            let key = match key {
+                key if key.is_instance_of::<PyUnicode>()? => key,
+                key if key.is_none() => intern!(py, "null"),
+                key if key.is_instance_of::<PyBool>()? => match key.is_true()? {
+                    true => intern!(py, "true"),
+                    false => intern!(py, "false"),
+                },
+                key => key.str()?,
```

Comment:
Here I'm (mostly) matching the behaviour of `json.dumps` that converts "dict keys of a basic type" (`str`, `int`, `float`, `bool`, `None`) to strings.

Again, this probably shouldn't live here, but I'm not clear what the right approach is.